feat: create landscape upgrade guide (26.04) (#204)
canonical/landscape-documentation: docs/how-to-guides/upgrade/upgrade-to-26-04-lts.md

---
myst:
  html_meta:
    description: "Upgrade Landscape Server to 26.04 LTS from Ubuntu 22.04, 24.04 or 26.04. Install additional outbox and repository mirroring services as snaps."
---

(how-to-upgrade-to-26-04-lts)=
# How to upgrade to Landscape Server 26.04 LTS

> 

Reviewer: ```suggestion
> <!-- TODO add when the release notes exist: See also: {ref}`reference-release-notes-26-04-lts` -->
```

```{important}

> If you are using repository mirroring in your Landscape deployment, you should not attempt to upgrade to 26.04 LTS at this time. A migration guide for bringing over repository mirrors from 24.04 LTS to 26.04 LTS will be published at a later date.

Reviewer: You should remove this as the note at the top, and put it in the intro (trying to have fewer long notes across the docs), also change text to be more in line with how we worded it in the Release notes - and I'd also recommend moving the note about the basic upgrade guide as its own section

So the beginning total will be like (I made some minor wording changes too 😇 ):

"
# How to upgrade to Landscape Server 26.04 LTS

You must be running Ubuntu 26.04 LTS Resolute Raccoon, 24.04 Noble Numbat, or 22.04 Jammy Jellyfish to upgrade to Landscape 26.04 LTS. 

Note that Quickstart installations and upgrades to Landscape 26.04 LTS **are not supported on Ubuntu 26.04.** You must use Ubuntu 24.04 LTS or 22.04 LTS for Quickstart installations.

If you use repository management in your Landscape deployment, we recommend waiting to upgrade to 26.04 LTS until the 26.04.1 point release (expected August 2026). A migration guide for bringing over repository mirrors from 24.04 LTS to 26.04 LTS will be published before the point release.

## Begin your upgrade

To upgrade your self-hosted Landscape server to 26.04 LTS, first follow the basic upgrade instructions. See {ref}`how-to-upgrade`.
"

--
I like the format of having the "To upgrade..." part first like you originally had it, but since we have so many conditions for this release, I want to communicate those first. LMK if you have thoughts on this structure though!

```

```{important}
Quickstart installations and upgrades to Landscape 26.04 LTS are not supported on Ubuntu 26.04.
```

To upgrade your self-hosted Landscape server to 26.04 LTS, you should first follow the basic upgrade instructions. See {ref}`how-to-upgrade`.

Note that you must be running Ubuntu 26.04 LTS Resolute Raccoon, 24.04 Noble Numbat, or 22.04 Jammy Jellyfish to upgrade to Landscape 26.04 LTS.

> 

Reviewer: Just noting that this would change based on my earlier suggestion

## Service.conf migration

> If you upgraded from a version prior to 25.10, your `service.conf` file will be migrated away from our deprecated naming system during the upgrade. A backup file from before the migration will be supplied in the same directory as your `service.conf`. Support for the deprecated config options is expected to be removed in the 26.10 release. See {ref}`reference-service-conf` for more details.

Author: https://github.com/canonical/landscape-documentation/pull/198 changes those

> If you upgraded from a version prior to 25.10, your `service.conf` file will be migrated away from our deprecated naming system during the upgrade. A backup file from before the migration will be supplied in the same directory as your `service.conf`. Support for the deprecated config options is expected to be removed in the 26.10 release. See {ref}`reference-service-conf` for more details.

Reviewer: I find this description confusing - can you say more directly what the user should do? It's not clear to me what the user should actually _do_ based on their version. It'd be better to structure it something like "If you're upgrading from 25.10, do XYZ. If you're upgrading from 25.04 or earlier, do XYZ."

The guide is technically any "any" upgrade to 26.04, but we'd consider the LTS to LTS migration route as being the main path here (24.04 to 26.04)

In other words - this section describes when it should actually tell the user what to do

Reviewer: @jmucc Can you look into this one? I'm not sure what's supposed to change here, but it's confusing because this section doesn't actually tell the user to do anything - it just provides information to them.

LMK if you want to discuss that at all though

Author: Sorry I missed this comment. There is no action required here from the user. If they upgraded from a version prior to 25.10, their config will be automatically migrated and they shouldn't have to do anything about it. Maybe it's out of place in a `how-to`, and other docs reference this same concept. I wouldn't be opposed to removing it entirely.

Reviewer: @jmucc ok! Yea since there's no action and it's just a FYI - you can remove it. Whenever that's done just re-request my review and I can double check / approve :)

## Additional upgrade steps

> After you’ve completed the basic upgrade instructions, you need to make some additional manual changes to finish your upgrade.

Reviewer: ```suggestion
After you’ve completed the basic upgrade instructions and have the updated `service.conf` file, you need to make some additional manual changes to finish your upgrade.
```
Or something like that - depends on how you change the doc from my earlier suggestions

### Install the outbox snap

> The `landscape-outbox` snap is critical to proper functioning of Landscape 26.04 LTS. Install the `landscape-outbox` snap on the same machine as your Landscape Server installation.

Reviewer: "The `landscape-outbox` snap is critical to proper functioning of Landscape 26.04 LTS. " doesn't provide that much meaningful info - I suggest you either:

- Change it to briefly describe the snap/how it's used in 1 sentence (e.g., "The `landscape-outbox` snap [describe what it does]"
OR, if it's not clear or helpful (i.e., too technical) to describe what it does
- Remove that first sentence, just start with "Install the..."

```bash
sudo snap install landscape-outbox
```

`landscape-outbox` is configured to work automatically with an existing Landscape Server by default. Check that the outbox is active and functioning properly.

```bash
sudo snap services landscape-outbox
sudo snap logs landscape-outbox
```

### Install the debarchive snap

> The `landscape-debarchive` snap is required for repository mirroring from Landscape 26.04 LTS onwards. Follow the instructions in the [dedicated guide](/docs/how-to-guides/landscape-installation-and-setup/debarchive-repository-management.md)

Author: Waiting on https://github.com/canonical/landscape-documentation/pull/202

> The `landscape-debarchive` snap is required for repository mirroring from Landscape 26.04 LTS onwards. Follow the instructions in the [dedicated guide](/docs/how-to-guides/landscape-installation-and-setup/debarchive-repository-management.md)

Reviewer: For now though, just wrap the text you want in a TODO, and I'll remove that as my final step before publishing, like this: https://github.com/canonical/landscape-documentation/pull/200/changes#diff-f08e7ef69c6c22d47b7c40056e4af2931d93608717bc1a7c41b7d9ca84151c04R11

You should write it so all I have to do is remove the TODO / hidden text and markers. (I'll still test it to make sure the link works)

Because you need the RTD build to pass in order to merge the PR

> ### Enable hostagent services (optional)

Reviewer: ```suggestion
### (WSL only) Enable hostagent services
```
Since it's not really "optional" - it's more that it only applies to certain users

> The hostagent services (`landscape-hostagent-consumer` and `landscape-hostagent-messenger`) are required to manage WSL instances via Ubuntu Pro for WSL. See [the hostagent_consumer section](/reference/config/service-conf.md/#the-hostagent-consumer-section) and [the hostagent_messenger section](/reference/config/service-conf.md/#the-hostagent-messenger-section) if you'd like to configure these services. If not configured, these services will fail to start, but all Landscape activities unrelated to WSL will still function properly.

Reviewer: You have the following errors in your RTD build (you can see them locally with `make run` or `make html`):
```
/home/docs/checkouts/readthedocs.org/user_builds/canonical-landscape/checkouts/204/docs/how-to-guides/upgrade/upgrade-to-26-04-lts.md:47: WARNING: 'myst' cross-reference target not found: '/docs/how-to-guides/landscape-installation-and-setup/debarchive-repository-management.md'
/home/docs/checkouts/readthedocs.org/user_builds/canonical-landscape/checkouts/204/docs/how-to-guides/upgrade/upgrade-to-26-04-lts.md:51: WARNING: local id not found in doc 'reference/config/service-conf': 'the-hostagent-consumer-section'
/home/docs/checkouts/readthedocs.org/user_builds/canonical-landscape/checkouts/204/docs/how-to-guides/upgrade/upgrade-to-26-04-lts.md:51: WARNING: local id not found in doc 'reference/config/service-conf': 'the-hostagent-messenger-section'
```

But for the last two, to link to certain headers, you still need to do the ref target format, so you'd add a line above that header like "(heading-hostagent-messenger)=" and then use the {ref} target syntax like normal. This is more resilient to changing the header text, someone can still change the section name and the ref target remains the same

([link to our internal docs that gives the syntax](https://canonical-landscape-internal.readthedocs-hosted.com/en/latest/documentation/contribute-to-docs/#links-within-the-docs-crossrefs))

Reviewer: @jmucc you added the ref targets but forgot to change the links here to use them :) (not sure if you're working on it still - just commenting because it's marked as ready for review but the build is still failing)

Author: OH let me fix that now

> The hostagent services (`landscape-hostagent-consumer` and `landscape-hostagent-messenger`) are required to manage WSL instances via Ubuntu Pro for WSL. See [the hostagent_consumer section](/reference/config/service-conf.md/#the-hostagent-consumer-section) and [the hostagent_messenger section](/reference/config/service-conf.md/#the-hostagent-messenger-section) if you'd like to configure these services. If not configured, these services will fail to start, but all Landscape activities unrelated to WSL will still function properly.

Reviewer: ```suggestion
These steps are only needed for WSL users. The hostagent services (`landscape-hostagent-consumer` and `landscape-hostagent-messenger`) are required to manage WSL instances via Ubuntu Pro for WSL. Follow the steps in [the hostagent_consumer section](/reference/config/service-conf.md/#the-hostagent-consumer-section) and [the hostagent_messenger section](/reference/config/service-conf.md/#the-hostagent-messenger-section) to configure these services. 

If you don't configure the hostagent services, you won't be able to use WSL with Landscape. Other activities unrelated to WSL will still function properly.
```
To make it more direct/clear

## Airgapped installations
If your deployment does not have internet access, you must carry the snaps into your airgapped environment as part of the 26.04 installation process.

First, in an environment with internet access, download the snaps.

```bash
snap download landscape-outbox
snap download landscape-debarchive --edge
```

For each snap, a `.snap` file and a `.assert` file will be produced.

After transferring the files to the airgapped environment, install the snaps.

```bash
sudo snap ack landscape-outbox_*.assert
sudo snap install landscape-outbox_*.snap
sudo snap ack landscape-debarchive_*.assert
sudo snap install landscape-debarchive_*.snap
```

> Finally, verify the snaps are working properly according to the instructions above.

Reviewer: What steps are these referring to?

If by "above", you mean the section that's literally right above, then I'd suggest you either 

- remove this line, and add the contents in Line 65, so it's like "After transferring the files to the airgapped environment, install the snaps and verify they're working properly:"
OR
- separate out the code block about so it's install, and add the verify commands below this line

---
myst:
  html_meta:
    description: "Upgrade Landscape Server to 26.04 LTS from Ubuntu 22.04, 24.04, or 26.04. Install additional outbox and repository mirroring services as snaps."
---

(how-to-upgrade-to-26-04-lts)=
# How to upgrade to Landscape Server 26.04 LTS
<!-- TODO add when the release notes exist: See also: {ref}`reference-release-notes-26-04-lts` -->
You must be running Ubuntu 26.04 LTS Resolute Raccoon, 24.04 Noble Numbat, or 22.04 Jammy Jellyfish to upgrade to Landscape 26.04 LTS.

Note that Quickstart installations and upgrades to Landscape 26.04 LTS are not supported on Ubuntu 26.04. You must use Ubuntu 24.04 LTS or 22.04 LTS for Quickstart installations.

If you use repository management in your Landscape deployment, we recommend waiting to upgrade to 26.04 LTS until the 26.04.1 point release (expected August 2026). A migration guide for bringing over repository mirrors from 24.04 LTS to 26.04 LTS will be published before the point release.

## Begin your upgrade

To upgrade your self-hosted Landscape server to 26.04 LTS, you should first follow the basic upgrade instructions. See {ref}`how-to-upgrade`.

## Additional upgrade steps

> After you’ve completed the basic upgrade instructions and have the updated `service.conf` file, you need to make some additional manual changes to finish your upgrade.

Reviewer: ```suggestion
After you’ve completed the basic upgrade instructions, you need to make some additional manual changes to finish your upgrade.
```
Just noticed this too - but you should remove it since we took out that other section which references the service.conf. Right now it just introduces new information w/o a resolution to it
